src/login_param.rs

```diff
@@ -837,10 +831,17 @@ impl ConfiguredLoginParam {
                 ),
             )
             .await?;
-        context
-            .sql
-            .set_raw_config(Config::ConfiguredAddr.as_ref(), Some(&addr))
-            .await?;
+        if configured_addr.is_none() {
```

Comment:
This can race with itself, e.g. if two `set_config_ex()` calls are done in parallel for an unconfigured context. While this is unlikely, i'd suggest to add some `try_set_raw_config()` which does `INSERT OR IGNORE`. Extra locking it's better to avoid. Orphaned `transport` entry shouldn't be a problem i think.

Reply:
> The problem with this approach is that ConfiguredProvider should be set iff there was no ConfiguredAddr.

Not a problem, `try_set_raw_config()` can return bool (whether the insertion happened).

`save_to_transports_table()` won't be called in parallel directly of course because it's an internal function, but `set_config()` can be. Both calls would succeed then which doesn't look correct.

While there's no such problem for other config keys, maybe just document that `set_config()` mustn't be called in parallel for the same key? Then `save_to_transports_table()` wouldn't violate anything.